modules/core/src/v2/coins/xlm.ts

```diff
@@ -908,15 +908,18 @@ export class Xlm extends BaseCoin {
     _.forEach(trustlineOperations, op => {
       const opToken = this.getTokenNameFromStellarAsset(op.line);
       const tokenTrustline = _.find(txParams.trustlines, trustline => {
-        return trustline.token === opToken &&
-          ((_.isUndefined(trustline.limit) && op.limit === Xlm.maxTrustlineLimit) || // no limit was specified, max by default
-            trustline.limit === op.limit);
+        const noLimit = _.isUndefined(trustline.limit); // no limit was specified
+        const addTrustlineWithDefaultLimit = (trustline.action === 'add' && op.limit === Xlm.maxTrustlineLimit);
+        const removeTrustline = (trustline.action === 'remove' && op.limit === '0');
+        return (trustline.token === opToken &&
```

Comment:
Condition logic here is making my head hurt a bit, but I think it's correct. Consider bringing some of these expressions out into variables and simplifying the return so it's a bit easier to follow.

Reply:
I thought I was simplifying the condition in the `return` statement by extracting the other 3 into  `noLimit `, `addTrustlineWithDefaultLimit ` and `addTrustlineWithDefaultLimit `. I added more comments.

Reply:
Thanks, that helps clear it up a bit.